src/main/java/spoon/MavenLauncher.java

```diff
@@ -0,0 +1,348 @@
+				sourcePath = build.getTestSourceDirectory();
+			}
+			if (sourcePath == null) {
+				sourcePath = Paths.get(directory.getAbsolutePath(), "src", "test", "java").toString();
+			}
+			File source = new File(sourcePath);
+			if (source.exists()) {
+				output.add(source);
+			}
+			File generatedSource = Paths.get(directory.getAbsolutePath(), "target", "generated-test-sources").toFile();
+			if (generatedSource.exists()) {
+				output.add(generatedSource);
+			}
+			for (InheritanceModel module : modules) {
+				output.addAll(module.getTestDirectories());
+			}
+			return output;
+		}
+
+		/**
+		 * Extract the variable from a string
+		 */
+		private String extractVariable(String value) {
+			if (value.startsWith("$")) {
+				value = getProperty(value.substring(2, value.length() - 1));
+			}
+			return value;
+		}
+
+		/**
+		 * Get the list of dependencies available in the local maven repository
+		 * @return the list of  dependencies
+		 */
+		public List<File> getDependencies() {
+			Set<File> output = new HashSet<>();
+
+			// add the parent has a dependency
+			Parent parent = model.getParent();
+			if (parent != null) {
+				String groupId = parent.getGroupId().replace(".", "/");
+				String version = extractVariable(parent.getVersion());
+				String fileName = parent.getArtifactId() + "-" + version + ".jar";
+				Path depPath = Paths.get(m2RepositoryPath, groupId, parent.getArtifactId(), version, fileName);
+				File jar = depPath.toFile();
+				if (jar.exists()) {
+					output.add(jar);
+				}
+			}
+			List<Dependency> dependencies = model.getDependencies();
+			for (Dependency dependency : dependencies) {
+				String groupId = dependency.getGroupId().replace(".", "/");
+				String version = extractVariable(dependency.getVersion());
+				String fileName = dependency.getArtifactId() + "-" + version + ".jar";
+				// TODO: Check the scope of the dependency (local dependency is not handled)
+				Path depPath = Paths.get(m2RepositoryPath, groupId, dependency.getArtifactId(), version, fileName);
+				File jar = depPath.toFile();
+				if (jar.exists()) {
+					output.add(jar);
+				} else {
+					// if the a dependency is not found, uses the no classpath mode
+					getEnvironment().setNoClasspath(true);
+				}
+			}
+
+			for (InheritanceModel module : modules) {
+				output.addAll(module.getDependencies());
+			}
+			return new ArrayList<>(output);
+		}
+
+		/**
+		 * Get the value of a property
+		 * @param key the key of the property
+		 * @return the property value if key exists or null
+		 */
+		private String getProperty(String key) {
+			if ("project.version".equals(key)) {
+				if (model.getVersion() != null) {
+					return model.getVersion();
+				}
+			}
+			String value = model.getProperties().getProperty(key);
+			if (value == null) {
+				if (parent == null) {
+					return null;
+				}
+				return parent.getProperty(key);
+			}
+			return value;
+		}
+
+		/**
+		 * Get the source version of the project (default 7)
```

Comment:
Why java 7 by default? Java 8 seems to be more used: https://plumbr.eu/blog/java/java-version-and-vendor-data-analyzed-2017-edition

Reply:
Because the default compliance level in spoon is 7

Reply:
Ok, maybe we should change that I'll open an issue.

Reply:
And then do we have a constant value for this default compliance level?

Reply:
Of course no...